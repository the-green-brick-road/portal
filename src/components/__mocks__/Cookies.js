/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }             from '@mui/material';

function Cookies(props) {

    /* --------- Gather inputs --------- */
    const { color } = props;

    /* -------- Defining theme --------- */
    return (
        <Box style={{ minHeight:'30px',zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> Cookies </Typography>
        </Box>
    );

}

export default Cookies;
