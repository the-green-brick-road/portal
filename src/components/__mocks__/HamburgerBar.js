/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerBar component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height, isNegative, theme={} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }

    return (
        <Box style={{height:height, backgroundColor:color, width: '100%'}}>
            <Typography> HamburgerBar </Typography>
        </Box>
    );

}

export default HamburgerBar;
