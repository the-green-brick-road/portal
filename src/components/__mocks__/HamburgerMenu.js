/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerMenu component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerMenu(props) {

    /* --------- Gather inputs --------- */
    const { top , margin, isNegative, theme={} } = props;
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }
    const menuWidthString = `calc( 100vw - 2 * ${margin})`;

    return (
        <Box style={{ minHeight:'30px', width:menuWidthString, top:top, zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> HamburgerMenu </Typography>
        </Box>
    );

}

export default HamburgerMenu;