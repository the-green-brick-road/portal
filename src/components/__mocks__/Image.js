/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'blue' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Image;
