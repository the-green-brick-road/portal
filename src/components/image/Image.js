/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @22 may 2023
# Latest revision: 14 june 2023
# ---------------------------------------------------- */

/* React includes */
import { Profiler }              from 'react';

/* Portal includes */
import { useDesign, useLogging } from '../../providers';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { name, style={} }    = props;
    const { images }            = useDesign();
    const { onRender }          = useLogging();
    const componentName = 'Image';

    /* ---- Select image if loaded ----- */
    let data = {}
    if (name in images) { data = images[name] }
    const label_picture = `picture-${name}`

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Profiler id={componentName} onRender={onRender}>
            {(data !== {}) && (
                <picture style={style} data-testid={label_picture}>
                    { Object.entries(data).toReversed().map((item) => {

                        let result;
                        const localType = `image/${item[0]}`;
                        const localSizes = Object.entries(item[1]);
                        let localSrcset = '';
                        for (let i = 0; i < localSizes.length; i += 1) {
                            localSrcset = `${localSrcset} ${localSizes[i][1].img} ${localSizes[i][1].width} ,`;
                        }
                        if (item[0] !== 'default') {
                            result = (<source key={item[0]} srcSet={localSrcset} type={localType} />);
                        }
                        return result;
                    })}
                    <img src={data.default} style={style} alt={name}/>
                </picture>
            )}
        </Profiler>
    );
    /* eslint-enable padded-blocks */

}

export default Image;
