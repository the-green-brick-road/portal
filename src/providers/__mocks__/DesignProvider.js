/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Design provider mock
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @09 may 2023
# Latest revision: 09 may 2023
# ---------------------------------------------------- */

/* React includes */
import { createContext }       from 'react';

/* Material UI includes */
import { CssBaseline }                                 from '@mui/material';
import { ThemeProvider, createTheme }                  from '@mui/material/styles';

const context = createContext(null);

/* ----- Define provider values ---- */
var state = {
    isWebpSupported: true,
    isSliding:       false,
    isDarkMode:      false,
    images:          {},
    screen:          '',
    sizes:           {},
    setIsSliding:    jest.fn((value) => { state.isSliding = value }),
};

function DesignProvider(props) {

    const { children, isWebpSupported, screen, isDarkMode = false, isSliding = false, sizes, images = [], theme = {} } = props;

    /* ------ Defining behaviour ------- */
    var local_theme = {}
    if (Object.keys(theme).length !== 0) { local_theme = createTheme(theme); }

    state.isWebpSupported = isWebpSupported;
    state.screen          = screen;
    state.images          = {}
    state.sizes           = sizes;
    state.isSliding       = isSliding;
    state.isDarkMode      = isDarkMode;
    for (const image of images)
    {

        const rawext = image["raw"].split('.').pop();
        const webext = image["web"].split('.').pop();

        state.images[image.name] = {}
        state.images[image.name][rawext] = {}
        state.images[image.name][webext] = {}

        state.images[image.name][rawext]['small']           = {}
        state.images[image.name][rawext]['small']['img']    = require(`../../assets/small/${image.raw}`)
        state.images[image.name][rawext]['small']['width']  = `${sizes['small-width']}w`
        state.images[image.name][rawext]['medium']          = {}
        state.images[image.name][rawext]['medium']['img']   = require(`../../assets/medium/${image.raw}`)
        state.images[image.name][rawext]['medium']['width'] = `${sizes['medium-width']}w`
        state.images[image.name][rawext]['large']           = {}
        state.images[image.name][rawext]['large']['img']    = require(`../../assets/${image.raw}`)
        state.images[image.name][rawext]['large']['width']  = `${sizes['large-width']}w`

        state.images[image.name][webext]['small']           = {}
        state.images[image.name][webext]['small']['img']    = require(`../../assets/small/${image.web}`)
        state.images[image.name][webext]['small']['width']  = `${sizes['small-width']}w`
        state.images[image.name][webext]['medium']          = {}
        state.images[image.name][webext]['medium']['img']   = require(`../../assets/medium/${image.web}`)
        state.images[image.name][webext]['medium']['width'] = `${sizes['medium-width']}w`
        state.images[image.name][webext]['large']           = {}
        state.images[image.name][webext]['large']['img']    = require(`../../assets/${image.web}`)
        state.images[image.name][webext]['large']['width']  = `${sizes['large-width']}w`

        state.images[image.name]['default']= require(`../../assets/${image.raw}`)

    }

    /* ----------- Define HTML --------- */
    return (
        <ThemeProvider theme={local_theme}>
            <CssBaseline />
            <context.Provider value={state}>
                {children}
            </context.Provider>
        </ThemeProvider>
    );

}

function useDesign() { return state; };

export { useDesign, DesignProvider };