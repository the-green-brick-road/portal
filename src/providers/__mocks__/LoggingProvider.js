/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Logging provider mock
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @09 may 2023
# Latest revision: 09 may 2023
# ---------------------------------------------------- */

/* React includes */
import { createContext }       from 'react';

const context = createContext(null);

/* ----- Define provider values ---- */
var state = {
    isLoggingActivated : true,
    logText: jest.fn((component, level, topic, message) => {  console.log(`${component} - ${level} - ${topic} -${message}`) }),
};

function LoggingProvider(props) {

    const { children } = props;

    /* ----------- Define HTML --------- */
    return (
        <context.Provider value={state}>
            {children}
        </context.Provider>
    );

}

function useLogging() { return state; };

export { useLogging, LoggingProvider };