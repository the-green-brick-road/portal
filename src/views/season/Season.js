/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Season page definition
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @30 may 2023
# Latest revision: 30 may 2023
# ---------------------------------------------------- */

/* React includes */
import { Fragment }  from 'react';

/* Portal includes */
import { useDesign } from '../../providers';

/* Local includes */
import SeasonDesktop from './SeasonDesktop';
import SeasonMobile  from './SeasonMobile';

function Season(props) {

    /* --------- Gather inputs --------- */
    const { data }      = props
    const { screen }    = useDesign();
    //const componentName = 'Season';

    /* ----------- Define HTML --------- */
    return (
        <Fragment>
            {(screen === 'large') && (<SeasonDesktop data={data} />)}
            {(screen !== 'large') && (<SeasonMobile data={data} />)}
        </Fragment>
    );

}

export default Season;