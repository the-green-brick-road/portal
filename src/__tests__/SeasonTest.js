/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Season view component test suite
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @30 may 2023
# Latest revision: 30 may 2023
# ---------------------------------------------------- */

/* Test includes */
import { render, prettyDOM }       from '@testing-library/react'
import { expect, test}             from '@jest/globals';

/* Component under test */
import { default as Season }       from '../views/season/Season';

/* Mocks includes */
/* eslint-disable jest/no-mocks-import */
import { default as MockImage }    from '../components/__mocks__/Image';
import { useDesign as mockUseDesign, DesignProvider as MockDesignProvider } from '../providers/__mocks__/DesignProvider';
import { useLogging as mockUseLogging, LoggingProvider as MockLoggingProvider } from '../providers/__mocks__/LoggingProvider';
jest.mock("../components", () => ({ Image: (props) => MockImage(props) }));
jest.mock('../providers', () => ({
    useDesign: (() => { return mockUseDesign(); }),
    useLogging: (() => { return mockUseLogging(); }),
}));
/* eslint-enable jest/no-mocks-import */

describe("Season view" ,() => {

    const sizes = {
        "small-width"          : 200,
        "medium-width"         : 400,
        "large-width"          : 800,
        "menu-height"          : 80,
        "margin"               : 10,
        "hamburger-height"     : 40,
    }

    test('Should display season page for desktop', async () => {

        const view = render(
            <div>
                <MockLoggingProvider>
                    <MockDesignProvider screen="large" sizes={sizes}>
                        <Season data={{
                            'name':'season',
                            'image':'https://th.bing.com/th/id/R.4aa108082e7d3cbd55add79f84612aaa?rik=I4dbPhSe%2fbHHSg&riu=http%3a%2f%2fpurepng.com%2fpublic%2fuploads%2flarge%2fpurepng.com-google-logo-2015brandlogobrand-logoiconssymbolslogosgoogle-6815229372333mqrr.png&ehk=ewmaCOvP0Ji4QViEJnxSdlrYUrTSTWhi8nZ9XdyCgAI%3d&risl=&pid=ImgRaw&r=0',
                            'reveal':'JBSUgDxICg8',
                            'robot-game':'T2sv8jXoP4s',
                            'robot-game-gbr':'T2sv8jXoP4s',
                            'judging-session-gbr':'temp.pdf',
                            'resources':[{'name':'google','url':'www.google.com'}],
                            'competitions' : [
                                {'name':'compet1', 'level':'regional','rank':'Champion','award':'best','date':{seconds:40000000}},
                                {'name':'compet2', 'level':'regional','rank':'Champion','award':'best','date':{seconds:60000000}},
                                {'name':'compet3', 'level':'international','rank':'Champion','award':'best','date':{seconds:50000000}},
                                {'name':'compet4', 'level':'international','rank':'Champion','award':'best','date':{seconds:45000000}}
                            ],
                        }}/>
                    </MockDesignProvider>
                </MockLoggingProvider>
            </div>
        );
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false, highlight: false});
        expect(tree).toMatchSnapshot();

    })

    test('Should display season page for mobile', async () => {

        const view = render(
            <div>
                <MockLoggingProvider>
                    <MockDesignProvider screen="small" sizes={sizes}>
                        <Season data={{
                            'name':'season',
                            'image':'https://th.bing.com/th/id/R.4aa108082e7d3cbd55add79f84612aaa?rik=I4dbPhSe%2fbHHSg&riu=http%3a%2f%2fpurepng.com%2fpublic%2fuploads%2flarge%2fpurepng.com-google-logo-2015brandlogobrand-logoiconssymbolslogosgoogle-6815229372333mqrr.png&ehk=ewmaCOvP0Ji4QViEJnxSdlrYUrTSTWhi8nZ9XdyCgAI%3d&risl=&pid=ImgRaw&r=0',
                            'reveal':'JBSUgDxICg8',
                            'robot-game':'T2sv8jXoP4s',
                            'robot-game-gbr':'T2sv8jXoP4s',
                            'judging-session-gbr':'temp.pdf',
                            'resources':[{'name':'google','url':'www.google.com'}],
                            'competitions' : [
                                {'name':'compet1', 'level':'regional','rank':'Champion','award':'best','date':{seconds:40000000}},
                                {'name':'compet2', 'level':'regional','rank':'Champion','award':'best','date':{seconds:60000000}},
                                {'name':'compet3', 'level':'international','rank':'Champion','award':'best','date':{seconds:50000000}},
                                {'name':'compet4', 'level':'international','rank':'Champion','award':'best','date':{seconds:45000000}}
                            ],
                        }}/>
                    </MockDesignProvider>
                </MockLoggingProvider>
            </div>
        );
        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false, highlight: false});
        expect(tree).toMatchSnapshot();

    })

})
