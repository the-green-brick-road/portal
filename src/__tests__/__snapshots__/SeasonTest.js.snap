// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Season view Should display season page for desktop 1`] = `
"<body>
  <div>
    <div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root\\"
        style=\\"pointer-events: none; width: 100%; height: 30vh; padding: 0px; background-color: rgb(0, 0, 0); position: relative;\\"
      >
        <img
          alt=\\"season\\"
          src=\\"https://th.bing.com/th/id/R.4aa108082e7d3cbd55add79f84612aaa?rik=I4dbPhSe%2fbHHSg&riu=http%3a%2f%2fpurepng.com%2fpublic%2fuploads%2flarge%2fpurepng.com-google-logo-2015brandlogobrand-logoiconssymbolslogosgoogle-6815229372333mqrr.png&ehk=ewmaCOvP0Ji4QViEJnxSdlrYUrTSTWhi8nZ9XdyCgAI%3d&risl=&pid=ImgRaw&r=0\\"
          style=\\"position: absolute; z-index: 1; top: 11vh; right: 2%; height: 15vh;\\"
        />
      </div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root\\"
        style=\\"padding: 10px;\\"
      >
        <div
          class=\\"MuiBox-root css-0\\"
        >
          <div
            class=\\"MuiTabs-root css-1ujnqem-MuiTabs-root\\"
          >
            <div
              class=\\"MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller\\"
              style=\\"overflow: hidden; margin-bottom: 0px;\\"
            >
              <div
                class=\\"MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer\\"
                role=\\"tablist\\"
              >
                <button
                  aria-selected=\\"true\\"
                  class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root\\"
                  id=\\"0\\"
                  role=\\"tab\\"
                  style=\\"font-weight: bold; text-transform: capitalize;\\"
                  tabindex=\\"0\\"
                  type=\\"button\\"
                >
                  Season Reveal
                  <span
                    class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                  />
                </button>
                <button
                  aria-selected=\\"false\\"
                  class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root\\"
                  id=\\"1\\"
                  role=\\"tab\\"
                  style=\\"font-weight: bold; text-transform: capitalize;\\"
                  tabindex=\\"-1\\"
                  type=\\"button\\"
                >
                  Resources
                  <span
                    class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                  />
                </button>
                <button
                  aria-selected=\\"false\\"
                  class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root\\"
                  id=\\"2\\"
                  role=\\"tab\\"
                  style=\\"font-weight: bold; text-transform: capitalize;\\"
                  tabindex=\\"-1\\"
                  type=\\"button\\"
                >
                  Robot Game
                  <span
                    class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                  />
                </button>
              </div>
              <span
                class=\\"MuiTabs-indicator css-1aquho2-MuiTabs-indicator\\"
                style=\\"left: 0px; width: 0px;\\"
              />
            </div>
          </div>
        </div>
        <div
          class=\\"\\"
          style=\\"padding-top: 20px; padding-bottom: 20px; text-align: center;\\"
        >
          <div
            class=\\"\\"
            id=\\"\\"
          />
        </div>
      </div>
    </div>
  </div>
</body>"
`;

exports[`Season view Should display season page for mobile 1`] = `
"<body>
  <div>
    <div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root\\"
        id=\\"season-mobile\\"
        style=\\"width: 100%; height: 30vh; padding: 0px; background-color: rgb(0, 0, 0); position: relative;\\"
      >
        <img
          alt=\\"season\\"
          src=\\"https://th.bing.com/th/id/R.4aa108082e7d3cbd55add79f84612aaa?rik=I4dbPhSe%2fbHHSg&riu=http%3a%2f%2fpurepng.com%2fpublic%2fuploads%2flarge%2fpurepng.com-google-logo-2015brandlogobrand-logoiconssymbolslogosgoogle-6815229372333mqrr.png&ehk=ewmaCOvP0Ji4QViEJnxSdlrYUrTSTWhi8nZ9XdyCgAI%3d&risl=&pid=ImgRaw&r=0\\"
          style=\\"position: absolute; z-index: 1; top: 11vh; right: 2%; height: 15vh;\\"
        />
      </div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root\\"
        style=\\"padding: 10px;\\"
      >
        <div
          class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1yof2we-MuiPaper-root-MuiAccordion-root\\"
        >
          <div
            aria-expanded=\\"false\\"
            class=\\"MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1ig3gh6-MuiButtonBase-root-MuiAccordionSummary-root\\"
            col=\\"#1976d2\\"
            role=\\"button\\"
            tabindex=\\"0\\"
          >
            <div
              class=\\"MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content\\"
            >
              <p
                class=\\"MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root\\"
                style=\\"font-weight: bold; font-size: 12px;\\"
              >
                Season Reveal
              </p>
            </div>
            <div
              class=\\"MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper\\"
            >
              <svg
                aria-hidden=\\"true\\"
                class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                data-testid=\\"ExpandMoreIcon\\"
                focusable=\\"false\\"
                style=\\"margin: 0px;\\"
                viewBox=\\"0 0 24 24\\"
              >
                <path
                  d=\\"M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z\\"
                />
              </svg>
            </div>
          </div>
          <div
            class=\\"MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root\\"
            style=\\"min-height: 0px;\\"
          >
            <div
              class=\\"MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper\\"
            >
              <div
                class=\\"MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner\\"
              >
                <div
                  class=\\"MuiAccordion-region\\"
                  role=\\"region\\"
                >
                  <div
                    class=\\"MuiAccordionDetails-root css-yuqx6y-MuiAccordionDetails-root\\"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1yof2we-MuiPaper-root-MuiAccordion-root\\"
        >
          <div
            aria-expanded=\\"false\\"
            class=\\"MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1ig3gh6-MuiButtonBase-root-MuiAccordionSummary-root\\"
            col=\\"#1976d2\\"
            role=\\"button\\"
            tabindex=\\"0\\"
          >
            <div
              class=\\"MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content\\"
            >
              <p
                class=\\"MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root\\"
                style=\\"font-weight: bold; font-size: 12px;\\"
              >
                Resources
              </p>
            </div>
            <div
              class=\\"MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper\\"
            >
              <svg
                aria-hidden=\\"true\\"
                class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                data-testid=\\"ExpandMoreIcon\\"
                focusable=\\"false\\"
                style=\\"margin: 0px;\\"
                viewBox=\\"0 0 24 24\\"
              >
                <path
                  d=\\"M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z\\"
                />
              </svg>
            </div>
          </div>
          <div
            class=\\"MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root\\"
            style=\\"min-height: 0px;\\"
          >
            <div
              class=\\"MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper\\"
            >
              <div
                class=\\"MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner\\"
              >
                <div
                  class=\\"MuiAccordion-region\\"
                  role=\\"region\\"
                >
                  <div
                    class=\\"MuiAccordionDetails-root css-yuqx6y-MuiAccordionDetails-root\\"
                  >
                    <table
                      class=\\"MuiTable-root css-rqglhn-MuiTable-root\\"
                      style=\\"width: 100%;\\"
                    >
                      <tbody
                        class=\\"MuiTableBody-root css-apqrd9-MuiTableBody-root\\"
                      >
                        <tr
                          class=\\"MuiTableRow-root css-1q1u3t4-MuiTableRow-root\\"
                        >
                          <td
                            class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                            style=\\"width: 0%; padding: 0px; border-style: none;\\"
                          >
                            <a
                              class=\\"MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-101ca9i-MuiTypography-root-MuiLink-root\\"
                              href=\\"www.google.com\\"
                              style=\\"color: black; text-decoration: none;\\"
                              target=\\"_blank\\"
                            >
                              <svg
                                aria-hidden=\\"true\\"
                                class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                                data-testid=\\"OpenInBrowserIcon\\"
                                focusable=\\"false\\"
                                style=\\"color: rgb(25, 118, 210);\\"
                                viewBox=\\"0 0 24 24\\"
                              >
                                <path
                                  d=\\"M19 4H5c-1.11 0-2 .9-2 2v12c0 1.1.89 2 2 2h4v-2H5V8h14v10h-4v2h4c1.1 0 2-.9 2-2V6c0-1.1-.89-2-2-2zm-7 6-4 4h3v6h2v-6h3l-4-4z\\"
                                />
                              </svg>
                            </a>
                          </td>
                          <td
                            class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                            style=\\"text-align: left; padding-top: 0px; padding-bottom: 0px; padding-left: 10px; border-style: none;\\"
                          >
                            <a
                              class=\\"MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-101ca9i-MuiTypography-root-MuiLink-root\\"
                              href=\\"www.google.com\\"
                              style=\\"color: black; text-decoration: none; font-size: 11px;\\"
                              target=\\"_blank\\"
                            >
                              google
                            </a>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1yof2we-MuiPaper-root-MuiAccordion-root\\"
        >
          <div
            aria-expanded=\\"false\\"
            class=\\"MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1ig3gh6-MuiButtonBase-root-MuiAccordionSummary-root\\"
            col=\\"#1976d2\\"
            role=\\"button\\"
            tabindex=\\"0\\"
          >
            <div
              class=\\"MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content\\"
            >
              <p
                class=\\"MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root\\"
                style=\\"font-weight: bold; font-size: 12px;\\"
              >
                Robot Game
              </p>
            </div>
            <div
              class=\\"MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper\\"
            >
              <svg
                aria-hidden=\\"true\\"
                class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                data-testid=\\"ExpandMoreIcon\\"
                focusable=\\"false\\"
                style=\\"margin: 0px;\\"
                viewBox=\\"0 0 24 24\\"
              >
                <path
                  d=\\"M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z\\"
                />
              </svg>
            </div>
          </div>
          <div
            class=\\"MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root\\"
            style=\\"min-height: 0px;\\"
          >
            <div
              class=\\"MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper\\"
            >
              <div
                class=\\"MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner\\"
              >
                <div
                  class=\\"MuiAccordion-region\\"
                  role=\\"region\\"
                >
                  <div
                    class=\\"MuiAccordionDetails-root css-yuqx6y-MuiAccordionDetails-root\\"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>"
`;
